pkg/cmd/admin/prune/deployments.go

```diff
@@ -89,11 +90,20 @@ func NewCmdPruneDeployments(f *clientcmd.Factory, parentName, name string, out i
 			case true:
 				deploymentPruneFunc = func(deployment *kapi.ReplicationController) error {
 					describingPruneDeploymentFunc(deployment)
-					err := kclient.ReplicationControllers(deployment.Namespace).Delete(deployment.Name)
-					if err != nil {
-						return err
+					// If the deployment is failed we need to remove its deployer pods, too.
+					if deployutil.DeploymentStatusFor(deployment) == deployapi.DeploymentStatusFailed {
+						dpSelector := deployutil.DeployerPodSelector(deployment.Name)
+						deployers, err := kclient.Pods(deployment.Namespace).List(dpSelector, fields.Everything())
+						if err != nil {
+							return err
+						}
+						for _, pod := range deployers.Items {
+							if err := kclient.Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
+								return err
+							}
+						}
 					}
-					return nil
+					return kclient.ReplicationControllers(deployment.Namespace).Delete(deployment.Name)
```

Comment:
If we fail to delete any deployer pod it looks like we won't actually delete others (or the deployment itself)... is that okay?